homeassistant/components/wolflink/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""The Wolf SmartSet Service integration."""
+from _datetime import timedelta
+import logging
+
+from wolf_smartset.token_auth import InvalidAuth
+from wolf_smartset.wolf_client import WolfClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=1)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Wolf SmartSet Service component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Wolf SmartSet Service from a config entry."""
+    username = entry.data["username"]
+    password = entry.data["password"]
+    device_name = entry.data["device_name"]
+    _LOGGER.debug("Setting up wolflink integration for device: %s", device_name)
+    hub = WolfClient(username, password)
+
+    try:
+        systems = await hub.fetch_system_list()
+    except InvalidAuth:
+        _LOGGER.error("Could not set up wolflink integration due to wrong credentials")
+        return False
+
+    filtered_systems = [device for device in systems if device.name == device_name]
+    gateway_id = filtered_systems[0].gateway
+    device_id = filtered_systems[0].id
+    parameters = await fetch_parameters(hub, gateway_id, device_id)
+
+    async def async_update_data():
+        """Update all stored entities for Wolf SmartSet."""
+        try:
+            values = await hub.fetch_value(gateway_id, device_id, parameters)
+            return {v.value_id: v.value for v in values}
+        except Exception as exception:
```

Comment:
Please have the library raise a library specific exception here that we can catch.

Reply:
It does throw specific exceptions. Changed. I think that this should be enough.